channel/src/main/java/com/synopsys/integration/alert/channel/email2/EmailChannelMessageSender.java

```diff
@@ -0,0 +1,155 @@
+import com.synopsys.integration.alert.common.exception.AlertConfigurationException;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.alert.common.message.model.MessageResult;
+import com.synopsys.integration.alert.common.persistence.accessor.ConfigurationAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.EmailJobDetailsModel;
+import com.synopsys.integration.alert.descriptor.api.EmailChannelKey;
+import com.synopsys.integration.alert.processor.api.extract.model.project.ProjectMessage;
+import com.synopys.integration.alert.channel.api.ChannelMessageSender;
+
+@Component
+public class EmailChannelMessageSender implements ChannelMessageSender<EmailJobDetailsModel, EmailChannelMessageModel, MessageResult> {
+    public static final String FILE_NAME_SYNOPSYS_LOGO = "synopsys.png";
+    public static final String FILE_NAME_MESSAGE_TEMPLATE = "message_content.ftl";
+
+    private final EmailChannelKey emailChannelKey;
+    private final AlertProperties alertProperties;
+    private final EmailAddressGatherer emailAddressGatherer;
+    private final EmailAttachmentFileCreator emailAttachmentFileCreator;
+    private final FreemarkerTemplatingService freemarkerTemplatingService;
+    private final ConfigurationAccessor configurationAccessor;
+
+    @Autowired
+    public EmailChannelMessageSender(
+        EmailChannelKey emailChannelKey,
+        AlertProperties alertProperties,
+        EmailAddressGatherer emailAddressGatherer,
+        EmailAttachmentFileCreator emailAttachmentFileCreator,
+        FreemarkerTemplatingService freemarkerTemplatingService,
+        ConfigurationAccessor configurationAccessor
+    ) {
+        this.emailChannelKey = emailChannelKey;
+        this.alertProperties = alertProperties;
+        this.emailAddressGatherer = emailAddressGatherer;
+        this.emailAttachmentFileCreator = emailAttachmentFileCreator;
+        this.freemarkerTemplatingService = freemarkerTemplatingService;
+        this.configurationAccessor = configurationAccessor;
+    }
+
+    @Override
+    public MessageResult sendMessages(EmailJobDetailsModel emailJobDetails, List<EmailChannelMessageModel> emailMessages) throws AlertException {
+        EmailProperties emailProperties = new EmailProperties(retrieveGlobalEmailConfig());
+        EmailMessagingService emailMessagingService = new EmailMessagingService(emailProperties, freemarkerTemplatingService);
+
+        EmailAttachmentFormat attachmentFormat = EmailAttachmentFormat.getValueSafely(emailJobDetails.getAttachmentFileType());
+
+        for (EmailChannelMessageModel message : emailMessages) {
+            // FIXME this does not currently account for project-summary messages which do not include a source
```

Comment:
This is tracked internally and will be resolved in a subsequent PR.

Reply:
Can this be resolved now with the source field?

Reply:
Yes, once that PR is merged I will add it.